File: Python/1049.py
tipo1 = input()
tipo2 = input()
tipo3 = input()

if ("vertebrado" == tipo1):
	if ("ave" == tipo2):
		if ("carnivoro" == tipo3):
			print ("aguia")
		elif ("onivoro" == tipo3):
			print ("pomba")
	elif ("mamifero" == tipo2):	
		if ("herbivoro" == tipo3):
			print ("vaca")
		elif ("onivoro" == tipo3):
			print ("homem")
elif ("invertebrado" == tipo1):
	if ("inseto" == tipo2):
		if ("hematofago" == tipo3):
			print ("pulga")
		elif ("herbivoro" == tipo3):
			print ("lagarta")
	elif ("anelideo" == tipo2):
		if ("hematofago" == tipo3):
			print ("sanguessuga")
		elif ("onivoro" == tipo3):
			print ("minhoca")

File: Ruby/1049.rb
tipo1 = gets.chomp
tipo2 = gets.chomp
tipo3 = gets.chomp

case tipo1
when "vertebrado"
	case tipo2
	when "ave"
		case tipo3
		when "carnivoro"
			puts ("aguia")
		when "onivoro"
			puts ("pomba")
		end
	when "mamifero"	
		case tipo3
		when "herbivoro"
			puts ("vaca")
		when "onivoro"
			puts ("homem")
		end
	end
when "invertebrado"
	case tipo2
	when "inseto"
		case tipo3
		when "hematofago"
			puts ("pulga")
		when "herbivoro"
			puts ("lagarta")
		end
	when "anelideo"
		case tipo3
		when "hematofago"
			puts ("sanguessuga")
		when "onivoro"
			puts ("minhoca")
		end
	end
end
